astropy/convolution/convolve.py

```diff
@@ -442,12 +452,27 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
                       "equivalent to the convolve boundary='fill'.  There is "
                       "no FFT equivalent to convolve's "
                       "zero-if-kernel-leaves-boundary", AstropyUserWarning)
-        psf_pad = True
+        if psf_pad is None:
+            psf_pad = True
+        if fft_pad is None:
+            fft_pad = True
     elif boundary == 'fill':
         # create a boundary region at least as large as the kernel
-        psf_pad = True
+        if psf_pad is not None:
```

Comment:
Maybe check that `psf_pad is not None and psf_pad is not True` -- not clear one should get a warning for doing what would be done by default anyway.